component/TimeProvider/work/timeprovider/C/src/timeprovider.c

```diff
@@ -0,0 +1,50 @@
+/* Body file for function TimeProvider
+ * Generated by TASTE (kazoo/templates/skeletons/c-body/function.tmplt)
+ * You can edit this file, it will not be overwritten
+
+    !! IMPORTANT                                                        !!
+    !! When you modify your design (interface view), you must update    !!
+    !! the procedures corresponding to the provided interfaces in this  !!
+    !! file. The up-to-date signatures can be found in the header file. !!
+*/
+#include "timeprovider.h"
+#include <Hal.h>
+
+static asn1SccTIME_PROVIDER_TIME reference_time = 0;
+static asn1SccTIME_PROVIDER_TIME applied_reference_time = 0;
+static asn1SccTIME_PROVIDER_TIME_REFERENCE_STATUS time_reference_status = asn1SccTIME_PROVIDER_TIME_REFERENCE_STATUS_not_synchronized;
+
+void timeprovider_startup(void)
+{
+}
+
+void timeprovider_PI_init(void)
+{
+}
+
+void timeprovider_PI_get_current_elapsed_time(asn1SccTIME_PROVIDER_TIME *OUT_current_time)
+{
+	*OUT_current_time = Hal_GetElapsedTimeInNs() - applied_reference_time;
```

Comment:
OK, let's assume the following:
Hal time  == 10 seconds
We receive reference time for synchronization == 100
Second later (when Hal time == 11) we receive synchronize. Applied time == 100
Second later when we get time, we get 12 - 100 == -88. Is that expected? Shouldn't we get 101?